Userland/Libraries/LibJS/Runtime/TypedArray.cpp

```diff
@@ -93,6 +93,32 @@ static void initialize_typed_array_from_array_buffer(GlobalObject& global_object
     typed_array.set_byte_offset(offset);
     typed_array.set_array_length(new_byte_length.value() / element_size);
 }
+template<typename T, TypedArrayType type>
+static void initialize_typed_array_from_typed_array(GlobalObject& global_object, TypedArray<T, type>& typed_array_to, TypedArrayBase& typed_array_from)
+{
+    auto array_length = typed_array_from.array_length();
+    auto element_size = typed_array_to.element_size();
+
+    typed_array_to.set_array_length(array_length);
+    typed_array_to.set_viewed_array_buffer(ArrayBuffer::create(global_object, array_length * element_size));
+    typed_array_to.set_byte_length(typed_array_to.viewed_array_buffer()->byte_length());
+
+    for (u32 i = 0; i < array_length; i++) {
+        Value v;
+        switch (typed_array_from.array_type()) {
+#undef __JS_ENUMERATE
+#define __JS_ENUMERATE(ClassName, snake_name, PrototypeName, ConstructorName, ArrayType)             \
+    case TYPED_ARRAY_ENUM_TYPE(ArrayType):                                                           \
+        v = dynamic_cast<TypedArray<ArrayType, TYPED_ARRAY_ENUM_TYPE(ArrayType)>&>(typed_array_from) \
+                .get_by_index(i);                                                                    \
```

Comment:
`dynamic_cast` can be quite slow, so it's sad to do it for every step of the iteration.
Can we hoist the casting out of the loop?

Reply:
Agree, but not sure if there is a way to do it, since we can't have an outside variable of type `TypedArray<T>` or is there?

Reply:
now replaced it with static cast, so I think should be better